sdk/resource/resource.go

```diff
@@ -231,6 +232,18 @@ func Default() *Resource {
 		if defaultResource == nil {
 			defaultResource = &Resource{}
 		}
+
+		if x.Resource.Enabled() {
+			instanceIDResource, err := defaultServiceInstanceIDDetector{}.Detect(context.Background())
```

Comment:
Instead of running this after the detection above, I would recommend constructing a slice a detectors (e.g. with and without the `x.Resource.Enabled`) and then pass that slice to `Detect` (i.e. line 222). It will avoid this duplicate merge logic being added here.

Reply:
Good call, I changed it as proposed.